### PR TITLE
Add a script that generates dx.all.d.ts

### PR DESCRIPTION
**Pull request by viterobk**

### CODE REVIEW

**Comment by AlekseyMartynov** [comment]
@viterobk and me discussed and decided to completely rewrite everything 😄

File: .eslintrc [hunk 1]
```diff
@@ -414,6 +414,8 @@
                     "squarified",
                     "squarify",
                     "src",
+                    "stderr",
```
Comment (AlekseyMartynov) [inline .eslintrc]: Not used anymore

**Review by AlekseyMartynov — DISMISSED** [review]

File: build/generate-dts.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const shell = require("shelljs");
+const TOOL_COMMAND = "devextreme-internal-declaration-tools";
+
+if(!shell.which(TOOL_COMMAND)) {
+    shell.echo(`${TOOL_COMMAND} command is not available`);
+    shell.exit(0);
+}
+
+if(shell.exec(TOOL_COMMAND).code > 0) {
+    shell.exit(1);
```
Comment (IlyaKhD) [inline build/generate-dts.js]: Maybe `shell.exit(shell.exec(TOOL_COMMAND).code)` ?

File: package.json [hunk 3]
```diff
@@ -110,13 +111,17 @@
...
   "browserslist": [
     "last 2 versions",
     "ios > 8",
     "ie > 10",
     "> 1%"
   ],
-  "pre-commit": "lint-staged"
+  "pre-commit": [
+    "lint-staged",
+    "generate-dts"
```
Comment (IlyaKhD) [inline package.json]: How do you think about renaming it to `update-ts` ? (as well as the global command)

**Review by AlekseyMartynov — APPROVED** [review]

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**